### PR TITLE
feat: new inet functions

### PR DESCRIPTION
**Pull request by panga**
Currently, INET extension is missing functions for INET range operations, related issue https://github.com/duckdb/duckdb/issues/10947.  
However, this PR aims to add the subnet contains operations and other helper functions.

* Add missing `BUILD_INET` in `Makefile`.
* All new functions works with IPv4/IPv6 and contains tests.
* Add five new INET functions compatible with PostgreSQL (https://www.postgresql.org/docs/current/functions-net.html):

`netmask` Computes the network mask for the address's network.
```sql
netmask(inet '192.168.1.5/24') → 255.255.255.0
```

`network` Returns the network part of the address, zeroing out whatever is to the right of the netmask.
```sql
network(inet '192.168.1.5/24') → 192.168.1.0/24
```

`broadcast` Computes the broadcast address for the address's network.
```sql
broadcast(inet '192.168.1.5/24') → 192.168.1.255/24
```

`<<=` Is subnet contained by or equal to subnet?
```sql
inet '192.168.1.5/32' <<= inet '192.168.1.0/24' → true
```

`>>=` Does subnet contain or equal subnet?
```sql
inet '192.168.1.5/32' >>= inet '192.168.1.0/24' → false
```

Note: I'm not familiar with duckdb code and it is my first contribution.

### CODE REVIEW

**Review by chrisiou — CHANGES_REQUESTED** [review]
hey @panga, thanks a lot for opening this PR! Everything looks good.
Could you please include some non-happy path test cases like more complex scenarios or failure cases? 
To ensure comprehensive testing coverage and prevent issues from future changes, it'd be nice to cover the following scenarios across all the new functions such as:
- subnet mask scenarios:
    - a single host mask (`/32` for IPv4, `/128` for IPv6)
    - a subnet mask covering all addresses (`/0` for both IPv4 and IPv6)
    - invalid mask values (e.g. `NULL` input, or bigger number than 32)
- some edge cases:
    - `192.168.1.255/24` or `2001:0db8::/96`
    - `0.0.0.0/19` or `::42`
- invalid address input values or invalid mask lengths
    - `999.999.999.999/24`, the existence of characters or `NULL` values

To test unexpected/incorrect usage, you can use:
```
statement error
<statement>
----
<error_msg>
```

File: test/sql/inet/test_inet_functions.test [hunk 1]
```diff
@@ -30,6 +30,39 @@ SELECT family('127.0.0.1/17');
...
+query I
+SELECT netmask(INET '192.168.1.5/24')
+----
+255.255.255.0/24
+
+query I
+SELECT netmask(INET '192.168.1.5/16')
+----
+255.255.0.0/16
+
```
[the remark this answers is not in the document]
Comment (panga) [reply test/sql/inet/test_inet_functions.test:48]: The requested new test cases were added.

File: test/sql/inet/test_ipv6_inet_functions.test [hunk 2]
```diff
@@ -30,6 +30,39 @@ SELECT family('::ffff:127.0.0.1/17')
 ----
 6
```
[the remark this answers is not in the document]
Comment (panga) [reply test/sql/inet/test_ipv6_inet_functions.test:32]: Done.

File: test/sql/inet/test_inet_operations.test [hunk 3]
```diff
@@ -71,9 +71,21 @@ SELECT id, i, j FROM tbl JOIN tbl2 USING (id)
...
 3	255.255.255.255/31	127.0.0.1
 
-
 query IIII
 SELECT tbl.id, tbl2.id, i, j FROM tbl JOIN tbl2 ON (i=j) ORDER BY tbl.id
 ----
 1	3	127.0.0.1	127.0.0.1
 5	3	127.0.0.1	127.0.0.1
+
+# subset
```
[the remark this answers is not in the document]
Comment (panga) [reply test/sql/inet/test_inet_operations.test:80]: Done.

File: test/sql/inet/test_ipv6_inet_operations.test [hunk 4]
```diff
@@ -71,9 +71,21 @@ SELECT id, i, j FROM tbl JOIN tbl2 USING (id)
...
-
 query IIII
 SELECT tbl.id, tbl2.id, i, j FROM tbl JOIN tbl2 ON (i=j) ORDER BY tbl.id
 ----
 1	3	::1	::1
-5	3	::1	::1
\ No newline at end of file
+5	3	::1	::1
+
+# subset
```
[the remark this answers is not in the document]
Comment (panga) [reply test/sql/inet/test_ipv6_inet_operations.test:80]: Done.

**Comment by panga** [comment]
@chrisiou thank you so much for the review. I've added the requested test cases.
Please note the new functions rely on INET datatype and thus only operates on valid IP v4/v6.
You can find the type conversion tests with invalid ip/mask in the test_inet_type.test files.

**Review by chrisiou — APPROVED** [review]
Looks good to me, thank you @panga!

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**